### PR TITLE
Use macros for Docs tables de-duplication

### PR DESCRIPTION
**Pull request by ambitious-octopus**
Changing export tables in 6 different pages is dangerous and we have a lot of repeated code. What do you think about this solution? If you think it is interesting we can also apply it to all the other repeated tables!

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
This PR updates the documentation workflow and enhances the export functionality using macros to streamline content management.

### 📊 Key Changes
- **Install `mkdocs-macros-plugin`**: Added to CI dependencies 
- **Modify Docs Conditions**: Adjusts markdown file conditions in `build_docs.py`
- **Replace Export Tables with Macros**: Uses macros for export tables across multiple documentation files
- **Bibtex Fixes**: Corrects the formatting of bibliographic references

### 🎯 Purpose & Impact
- **Streamline Docs Maintenance**: 🖇️ Makes it easier to manage repetitive content in documentation, enhancing efficiency.
- **Fix Bibliographic Formatting**: 📚 Ensures correct citation formats, aiding in academic and professional referencing.
- **Modify Export Conditions**: 🔄 Simplifies the logic for adding frontmatter in specific directories, improving content generation logic.

### CODE REVIEW

**Comment by ambitious-octopus** [comment]
@glenn-jocher this PR introduces a new dependency for serving documentation: [mkdocs-macros-plugin](https://mkdocs-macros-plugin.readthedocs.io/en/latest/). I just updated the CI and `pyproject.toml`. This plugin enables us to convert our Markdown pages into Jinja2 templates, allowing insertion of Python variables into pages using the `{{ variable }}` syntax.           

Changes were also made to files like YOLO7 and YOLOv9 because BibTeX quotes use the same` {{ ... }}` syntax. To prevent conflicts, we need to tell Jinja2 to treat these quotes as raw text instead of variables.

**Comment by glenn-jocher** [comment]
@ambitious-octopus we have bibtex in literally a dozen pages, why have you modified 2?

File: docs/en/macros/export-args.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+comments: false
+description: Comprehensive table on using the export function for Ultralytics YOLO models with detailed argument specifications.
+keywords: machine learning, model export, ONNX, TorchScript, TensorFlow, Keras, optimization, quantization, dynamic input, TensorRT, CoreML, NMS, batch inference
+---
```
Comment (glenn-jocher) [inline docs/en/macros/export-args.md:5]: Will the action include all of this on every page?

File: docs/en/models/yolov7.md [hunk 2]
```diff
@@ -103,12 +103,14 @@ We would like to acknowledge the YOLOv7 authors for their significant contributi
     === "BibTeX"
 
         ```bibtex
+        {% raw %}
```
Comment (glenn-jocher) [inline docs/en/models/yolov7.md:106]: Why has this been applied here? Why are the other 10 pages with bibtex entries fine without it?

File: docs/en/models/yolov9.md [hunk 3]
```diff
@@ -189,12 +189,14 @@ We would like to acknowledge the YOLOv9 authors for their significant contributi
     === "BibTeX"
 
         ```bibtex
+        {% raw %}
```
Comment (glenn-jocher) [inline docs/en/models/yolov9.md:192]: Same as YOLOv7 comment. Why here and not elsewhere?

File: docs/en/models/yolov7.md (hunk 2 above)
Comment (ambitious-octopus) [reply docs/en/models/yolov7.md]: Actually looking closely the problem here is the citations in bibtex. These two pages had a wrong syntax. I regenerated the citations in bibtex with google scholar and the correct format is as follows (Without curly brackets around YOLOv7):
```bibtex
@article{wang2022yolov7,
  title={YOLOv7: Trainable bag-of-freebies sets new state-of-the-art for real-time object detectors},
  author={Wang, Chien-Yao and Bochkovskiy, Alexey and Liao, Hong-Yuan Mark},
  journal={arXiv preprint arXiv:2207.02696},
  year={2022}
  }}
```
Wrong:
```bibtex
@article{wang2022yolov7,
  title={{YOLOv7}: Trainable bag-of-freebies sets new state-of-the-art for real-time object detectors},
  author={Wang, Chien-Yao and Bochkovskiy, Alexey and Liao, Hong-Yuan Mark},
  journal={arXiv preprint arXiv:2207.02696},
  year={2022}
  }}
```

No need to add any raw text syntax here. Updated!

File: docs/en/models/yolov9.md (hunk 3 above)
Comment (ambitious-octopus) [reply docs/en/models/yolov9.md]: same as above

File: docs/en/macros/export-args.md (hunk 1 above)
Comment (ambitious-octopus) [reply docs/en/macros/export-args.md]: Yes, I updated the action to ignore pages under the macros folder! We don't want meta here!

**Review by ambitious-octopus — COMMENTED** [review]
Using the mkdocs-macros plugin we can define python objects in the meta field (how cool!). I added a variable to the task pages that changes the model name so the tables are updated at build time!

File: pyproject.toml [hunk 4]
```diff
@@ -94,6 +94,7 @@ dev = [
     "mkdocs-jupyter", # for notebooks
     "mkdocs-redirects", # for 301 redirects
     "mkdocs-ultralytics-plugin>=0.0.49", # for meta descriptions and images, dates and authors
+    "mkdocs-macros-plugin>=1.0.5"
```
[the remark this answers is not in the document]
Comment (glenn-jocher) [reply pyproject.toml]: Updated

File: docs/en/datasets/detect/lvis.md [hunk 5]
```diff
@@ -97,9 +97,9 @@ If you use the LVIS dataset in your research or development work, please cite th
 
     === "BibTeX"
 
-        ```bibtex
+        ```BibTeX
```
[the remark this answers is not in the document]
Comment (glenn-jocher) [reply docs/en/datasets/detect/lvis.md]: Resolved.

**Review by glenn-jocher — APPROVED** [review]

**Comment by glenn-jocher** [comment]
@ambitious-octopus PR is looking amazing now, merged. 

Nice work!!

**Pull request merged**